LanguageFeatures/Patterns/variable_A02_t02.dart

```diff
@@ -83,13 +83,13 @@ String testList1(List l, [bool doTypeTest = false]) {
 
 String testList2(List l) {
   return switch (l) {
-    [a, b] => "[$a, $b]",
+    var [a, b] => "[$a, $b]",
     _ => "default"
   };
 }
 
 String testList3(List l, [bool doTypeTest = false]) {
-  if (l case [a, b]) {
+  if (l case final [a, b]) {
```

Comment:
I'm not sure that the `final` (or `var`) keyword can be applied to the outer pattern. Both the CFE and the Analyzer report compile-time errors in this case, but accept the program if the keyword immediately precedes the variable names.

The following program is rejected by the tools.

```dart
test(dynamic x) {
  if (x case final [a, b]) {
    return a + b;
  }
  return null;
}
```

The following program is accepted by the tools.

```dart
test(dynamic x) {
  if (x case [final a, final b]) {
    return a + b;
  }
  return null;
}
```

Reply:
@chloestefantsova yes, you are right. After rereading the spec I found that the following
```dart
var (a, [b, c]) = ("str", [1, 2]);
```
is allowed in a local variable declaration pattern only. It's not allowed after the `case` statements. I'll update the test